### PR TITLE
[Feature][transform]flink transform support multi split

### PR DESCRIPTION
**Pull request by lcyyyyyy**
…tion name

## Purpose of this pull request

flink transform module support multi split,and add custome split function name 
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-transforms/seatunnel-transforms-flink/seatunnel-transform-flink-split/src/main/java/org/apache/seatunnel/flink/transform/Split.java [hunk 1]
```diff
@@ -94,6 +97,9 @@ public void prepare(FlinkEnvironment prepareEnv) {
         if (config.hasPath(SEPARATOR)) {
             separator = config.getString(SEPARATOR);
         }
+        if (config.hasPath(NAME)) {
+            name = config.getString(NAME);
+        }
```
Comment (CalvinKirs) [inline seatunnel-transforms/seatunnel-transforms-flink/seatunnel-transform-flink-split/src/main/java/org/apache/seatunnel/flink/transform/Split.java:102]: This is not referenced

Comment (lcyyyyyy) [reply seatunnel-transforms/seatunnel-transforms-flink/seatunnel-transform-flink-split/src/main/java/org/apache/seatunnel/flink/transform/Split.java:102]: > default split function name = split,but we can define name in config file,like this:
 Split{
   separator=" "
   fields = ["word","name"]
   name = "splitx"
  }
  Split{
     separator="i"
     fields = ["word","test"]
     name = "splity"
    }

Comment (CalvinKirs) [reply seatunnel-transforms/seatunnel-transforms-flink/seatunnel-transform-flink-split/src/main/java/org/apache/seatunnel/flink/transform/Split.java:102]: ok, thanks your explain

**Comment by Hisoka-X** [comment]
I don't understand what's the point of supporting changing the name of the Split function, is there something I haven't considered?

**Comment by lcyyyyyy** [comment]
> 
Seatunnel support multi source&sink, But when we use the Split function in flink-transform, we can only use one kind of separator ,like this:
sql = select split(raw_message) from source_table;
Now we can register different Spilt function with different separator ,Like the example I said above，
In order to distinguish them better, the names  of the Split function is needed

The reason is that I have multiple data sources and may need to set different separator

**Comment by lcyyyyyy** [comment]
> I don't understand what's the point of supporting changing the name of the Split function, is there something I haven't considered?

Seatunnel support multi source&sink, But when we use the Split function in flink-transform, we can only use one kind of separator ,like this:
sql = select split(raw_message) from source_table;
Now we can register different Spilt function with different separator ,Like the example I said above，
In order to distinguish them better, the names of the Split function is needed

The reason is that I have multiple data sources and may need to set different separator

**Comment by Hisoka-X** [comment]
> > I don't understand what's the point of supporting changing the name of the Split function, is there something I haven't considered?
> 
> Seatunnel support multi source&sink, But when we use the Split function in flink-transform, we can only use one kind of separator ,like this: sql = select split(raw_message) from source_table; Now we can register different Spilt function with different separator ,Like the example I said above， In order to distinguish them better, the names of the Split function is needed
> 
> The reason is that I have multiple data sources and may need to set different separator

I get you. In order that, you should add config for custom separator too.

**Comment by lcyyyyyy** [comment]
> > > I don't understand what's the point of supporting changing the name of the Split function, is there something I haven't considered?
> > 
> > 
> > Seatunnel support multi source&sink, But when we use the Split function in flink-transform, we can only use one kind of separator ,like this: sql = select split(raw_message) from source_table; Now we can register different Spilt function with different separator ,Like the example I said above， In order to distinguish them better, the names of the Split function is needed
> > The reason is that I have multiple data sources and may need to set different separator
> 
> I get you. In order that, you should add config for custom separator too.


SeaTunnel already support custom separator,default separator=" "
So，support change the name of the Split function ,will support register multi Split function with different separator&field
You can try the following flink.batch.conf，it worked 

env {
  # You can set flink configuration here
  execution.parallelism = 1
}

source {
  # This is a example input plugin **only for test and demonstrate the feature input plugin**
  FakeSource {
    result_table_name = "test"
    mock_data_size = 100
  }

  # If you would like to get more information about how to configure seatunnel and see full list of input plugins,
  # please go to https://seatunnel.apache.org/docs/flink/configuration/source-plugins/Fake
}

transform {
  Split{
   separator=" "
   fields = ["word","name"]
   name = "splitx"
  }
  Split{
     separator="i"
     fields = ["word","test"]
     name = "splity"
    }
  Sql {
    sql = "select splitx(name),name,splity(name) from test"
  }

}

sink {
  # choose stdout output plugin to output data to console
   ConsoleSink{}
}

**Comment by lcyyyyyy** [comment]
> > > I don't understand what's the point of supporting changing the name of the Split function, is there something I haven't considered?
> > 
> > 
> > Seatunnel support multi source&sink, But when we use the Split function in flink-transform, we can only use one kind of separator ,like this: sql = select split(raw_message) from source_table; Now we can register different Spilt function with different separator ,Like the example I said above， In order to distinguish them better, the names of the Split function is needed
> > The reason is that I have multiple data sources and may need to set different separator
> 
> I get you. In order that, you should add config for custom separator too.

SeaTunnel already support custom separator,default separator = " " 
So ,support change the name of the Split function,will bring the feature to support register multi Split function with different separator&field
You can try the following flink.batch.conf,it worked

env {
  execution.parallelism = 1
}

source {
  FakeSource {
    result_table_name = "test"
    mock_data_size = 100
  }
}

transform {
  Split{
   separator=" "
   fields = ["word","name"]
   name = "splitx"
  }
  Split{
     separator="i"
     fields = ["word","test"]
     name = "splity"
    }
  Sql {
    sql = "select splitx(name),splity(name) from test"
  }

}

sink {
   ConsoleSink{}
}

**Comment by Hisoka-X** [comment]
@lcyyyyyy Thanks, I missed that

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Pull request merged**